### DISCUSSION TITLE
Why are the positive samples generated twice on LinkNeighborSampler?

### DISCUSSION THREAD

**Discussion by daviddavo** [pyg-team/pytorch_geometric#8170] [Q&A]
Using `LinkNeighborLoader` on a Heterogeneous graph with binary negative sampling will return an edge store like this (batch size=8).

```
(user, vote, proposal)={
     edge_index=[2, 15],
     input_id=[8],
     edge_label_index=[2, 16],
     edge_label=[16],
   },
```

It makes sense that `edge_label_index` is twice the batch size, as it has 8 positive samples and 8 negative samples (which can be selected using `edge_label`). But then what is `edge_index` for? Why does it have 15 pairs? Why not 8? If I disable negative sampling, there would be exactly 8 samples (but `edge_label_index` still exists).

Are the "positive samples" generated twice?

**Comment by akihironitta** [comment]
I remember I encountered a similar issue and I think #7791 fixed it. Would you mind trying it again with [the master branch or nightly release](https://github.com/pyg-team/pytorch_geometric#nightly-and-master)?
